Added metadata examples for Video, Recipe, and Review types

nzukoski opened the pull request:
Also moves Article metadata into the same folder.

This pull request fixes #491

jmadler commented on examples/metadata-examples/README.md:
```diff
@@ -0,0 +1,18 @@
+  review-json-ld.amp.html       - JSON-LD structured data markup format for Reviews
+  review-microdata.amp.html     - microdata structured data markup format for Reviews
+  video-json-ld.amp.html       - JSON-LD structured data markup format for Videos
+  video-microdata.amp.html     - microdata structured data markup format for Videos
```
Can you vertically align the notes?

jmadler commented on examples/metadata-examples/video-json-ld.amp.html:
```diff
@@ -0,0 +1,89 @@
+  <body>
+    <h2>The Audience is programing</h2>
+    <p>A programming audience</p>
+    <amp-video
```
The src URL must be absolute (and https or schema non specific)

nzukoski replied:
Done.

jmadler commented on examples/metadata-examples/video-json-ld.amp.html:
```diff
@@ -0,0 +1,89 @@
+        "thumbnailUrl": "http://cdn.ampproject.org/leader.jpg",
+        "uploadDate": "2015-02-05T08:00:00+08:00",
+        "duration": "PT1M33S",
+        "contentUrl": "http://cdn.ampproject.org/examples/video/The-Audience-Is-Programming.mp4",
```
Use HTTPS please

jmadler replied:
(just for the contentUrl, to match the below)

jmadler commented on examples/metadata-examples/video-microdata.amp.html:
```diff
@@ -0,0 +1,67 @@
+<!doctype html>
```
Same comments as ld-json version

nzukoski replied:
Done.

jmadler commented:
LGTM - please squash and also handle CLA

nzukoski commented:
I signed the CLA

nzukoski commented:
Took me a little while to figure out the squashing but it should be good to go now.

jmadler commented:
Update example filenames in these two places:

https://github.com/ampproject/amphtml/blob/master/gulpfile.js#L298
https://github.com/ampproject/amphtml/blob/master/test/integration/test-example-validation.js#L37

nzukoski commented:
Fixed the example file names and also updated the mainEntityOfPage stuff for the json-ld examples. Not sure why the Travis CI build is still failing...

erwinmombay commented:
@nzukoski was just a flakey test. so the travis ci build failing isnt this PR's fault. do you might resubmitting to make it run again? thanks!

nzukoski commented:
All tests passing now.

jmadler commented:
LGTM
